app/models/proposal.rb

```diff
@@ -60,10 +60,7 @@ class Proposal < ActiveRecord::Base
   scope :for_state, ->(state) do
     where(state: state).order(:title).includes(:event, {speakers: :user}, :review_taggings)
   end
-  scope :in_track, ->(track) do
-    track = nil if track.try(:strip).blank?
-    where(track: track)
-  end
```

Comment:
Also, track.try(:strip) == '' should work for both string and integer ids I think? Whitespace strings would get converted to nil, and integer ids would be left alone. Doing that plus renaming the track param to track_id might be enough here.